pkg/plugins/policies/matchers/match_test.go

```diff
@@ -0,0 +1,110 @@
+package matchers_test
+
+import (
+	"bytes"
+	"os"
+	"path"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	kubectl_output "github.com/kumahq/kuma/app/kumactl/pkg/output/yaml"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/model/rest"
+	_ "github.com/kumahq/kuma/pkg/plugins/policies"
+	"github.com/kumahq/kuma/pkg/plugins/policies/matchers"
+	policies_api "github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/api/v1alpha1"
+	test_matchers "github.com/kumahq/kuma/pkg/test/matchers"
+	"github.com/kumahq/kuma/pkg/util/yaml"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+)
+
+var _ = Describe("Match", func() {
+
+	readDPP := func(file string) *core_mesh.DataplaneResource {
+		dppYaml, err := os.ReadFile(path.Join("testdata", file))
+		Expect(err).ToNot(HaveOccurred())
+
+		dpp, err := rest.YAML.UnmarshalCore(dppYaml)
+		Expect(err).ToNot(HaveOccurred())
+		return dpp.(*core_mesh.DataplaneResource)
+	}
+
+	readResourceListResponse := func(file string) xds_context.Resources {
+		responseBytes, err := os.ReadFile(path.Join("testdata", file))
+		Expect(err).ToNot(HaveOccurred())
+
+		rawResources := yaml.SplitYAML(string(responseBytes))
+		resourceList := &policies_api.MeshTrafficPermissionResourceList{}
+		for _, rawResource := range rawResources {
+			resource, err := rest.YAML.UnmarshalCore([]byte(rawResource))
+			Expect(err).ToNot(HaveOccurred())
+			err = resourceList.AddItem(resource)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		return xds_context.Resources{
+			MeshLocalResources: map[core_model.ResourceType]core_model.ResourceList{
+				policies_api.MeshTrafficPermissionType: resourceList,
+			},
+		}
+	}
+
+	printPolicies := func(policies []core_model.Resource) string {
+		matchedPolicyList := &policies_api.MeshTrafficPermissionResourceList{}
+		for _, policy := range policies {
+			Expect(matchedPolicyList.AddItem(policy)).To(Succeed())
+		}
+		bytesBuffer := &bytes.Buffer{}
+		err := kubectl_output.NewPrinter().Print(rest.From.ResourceList(matchedPolicyList), bytesBuffer)
+		Expect(err).ToNot(HaveOccurred())
+		return bytesBuffer.String()
+	}
```

Comment:
personal nit: I'd inline all those functions. They are used once. At least for me, it's easier to follow the test if I don't have to jump back and forth between the test and local functions.